quinn-proto/src/connection/mod.rs

```diff
@@ -3309,6 +3362,12 @@ impl Connection {
         self.path.mtud.on_peer_max_udp_payload_size_received(
             u16::try_from(self.peer_params.max_udp_payload_size.into_inner()).unwrap_or(u16::MAX),
         );
+        self.ack_timestamps_cfg = params.ack_timestamps_cfg;
+        if let Some(max) = params.ack_timestamps_cfg.max_timestamps_per_ack {
+            for space in self.spaces.iter_mut() {
+                space.pending_acks.set_receiver_timestamp(max.0 as usize);
```

Comment:
Clamp this maximum to a reasonable value to guard against memory exhaustion attacks.